src/substrate/filters/enricher.ts

```diff
@@ -101,55 +110,103 @@ export async function Enrich(
+
+          const commissionPer =  (Number)(preference.commission || new BN(0)) / 10_000_000;
+
+          const rewardDestination = await api.query.staking.payee.at(hash,key);
+          const controllerId = await api.query.staking.bonded.at(hash, key);
+          let nextSessionKeysOpt;
+
+          try{
+            // to get keys of old blocks
+            nextSessionKeysOpt = await api.query.session.nextKeys(hash,key);
+            // there is queuedKeys function as well not sure which one to use
+            // nextSessionKeysOpt = await api.query.session.queuedKeys(hash,key);
+          }
+          catch(e){
+            // for new blocks after 3139200
+            nextSessionKeysOpt = await api.query.session.nextKeys.at(hash,key);
+          }
+          
+          const nextSessionKeys = nextSessionKeysOpt.isSome
+            ? nextSessionKeysOpt.unwrap()
+            : []
+
           validatorInfo[key] = {
             commissionPer,
-            controllerId,
-            rewardDestination,
-            nextSessionIds: await (await api.query.session.nextKeys(key)).toString(),
-            eraPoints: Number(eraPointsIndividual[key])
+            controllerId: controllerId.toString() == ''? key: controllerId.toString(),
+            rewardDestination: rewardDestination.toString(),
+            nextSessionIds: nextSessionKeys.map(key => key.toString()),
+            eraPoints: eraPointsIndividual[key]? Number(eraPointsIndividual[key]): 0
           };
-        });
-        // erasStakers(EraIndex, AccountId): Exposure -> api.query.staking.erasStakers // KUSAMA
-        // stakers(AccountId): Exposure -> api.query.staking.stakers // EDGEWARE
-        const stakersCall = (api.query.staking.stakers)
-          ? api.query.staking.stakers
-          : api.query.staking.erasStakers;
-        const stakersCallArgs = (account) => (api.query.staking.stakers)
-          ? [account]
-          : [currentEra as unknown as EraIndex, account];
+        };
+
+        const stakersCall = (account) => {
+          return  api.query.staking.stakers ?
+          api.query.staking.stakers.at(hash, account) : 
+          api.query.staking.erasStakers.at(hash,currentEra, account);
+        }
+
         let activeExposures: { [key: string]: any } = {}
         if (validators && currentEra) { // if currentEra isn't empty
-          active = validators.validators;
-          waiting = validators.nextElected;
-          await Promise.all(active.map(async (validator) => {
-            const tmp_exposure = (await stakersCall(...stakersCallArgs(validator))) as Exposure;
+          await Promise.all(validators.map(async (validator) => {
+            const tmpExposure = (await stakersCall(validator)) as Exposure;
             const exposure = {
-              own: tmp_exposure.own,
-              total: tmp_exposure.total,
-              others: tmp_exposure.others
+              own: tmpExposure.own,
```

Comment:
Since we know exactly what all the fields are, why are we using the `any` type?

Additionally, we're maintaining Codec types here, when we should be converting to known types.

This should also be fixed in `types.ts` -- the interface also specifies `any`.

Reply:
created interface for active exposures.